src/sentry/grouping/ingest/hashing.py

```diff
@@ -340,3 +342,22 @@ def get_hash_values(
         job["finest_tree_label"] = all_hashes.finest_tree_label
 
     return (primary_hashes, secondary_hashes, all_hashes)
+
+
+def get_or_create_grouphashes(
+    project: Project, calculated_hashes: CalculatedHashes
+) -> list[GroupHash]:
+    grouphashes = []
+
+    for hash_value in extract_hashes(calculated_hashes):
+        grouphash, created = GroupHash.objects.get_or_create(project=project, hash=hash_value)
+
+        # TODO: Do we want to expand this to backfill metadata for existing grouphashes? If we do,
+        # we'll have to override the metadata creation date for them.
+        if created and options.get("grouping.grouphash_metadata.ingestion_writes_enabled"):
```

Comment:
do we want to roll this out more slowly if we think there's going to be a hit in ingest DB performance (ie. in s4s first)?

Reply:
Josh and I agreed that as long as there's a killswitch we're fine. Keep in mind that for now this is just new hashes, which only come either during a config transition or with new groups (which happen in < 1% of events).